# Phase 2 Project

    Student name: Phase 2 Group 8 (Damaris Wanjiru / Henry Kemboi / Elizabeth Atieno / Gerald Muhia / Caleb Kipkech / Collins Kwata)
    Student pace: Full Time Remote
    Scheduled project review date/time:
    Instructor name: Asha Deen
    

# Project Overview

Our company is launching a new film studio to tap into the growing demand for original video content. With no prior experience in the industry, we have been tasked with analyzing current box office trends to identify the most successful film types. The goal is to deliver actionable insights that will guide the studio's direction and ensure a strong start in the competitive movie market.

# Business Understanding
* **Business Context**: Our company aims to enter the original video content market by launching a new film studio, despite having no experience in movie production. To ensure success, we are tasked with analyzing current box office trends to identify top-performing film types. Our insights will guide the studio's leadership in making informed decisions about which genres to focus on.
*  **Problem Statement**: We need to identify top-performing film type, understand the current box office trends and evaluate the impact of different factors on movie perfomance.

# Objectives
1. Analyze the correlation between runtime and genre tand whether either of the two has an impact on average ratings.
2. Evaluate the impact of release date on genre ratings to identify trends and pinpoint which genres consistently receive high ratings over time.
3. Analyze audience preferences across different MPAA ratings to understand how each rating category appeals to various demographics.4Identify the genre that 
4. Assess directors' performance to determine which has achieved the highest return on investment within specific genres.
5. Identify directors known for producing critically acclaimed works within top-rated genres.
6. Determine which movie has achieved the highest return on investment, considering both production costs and box office earnings.


### Importing the Relevant Libraries

To begin, we will import all the relevant libraries necessary for our analysis. These are listed below:

- `Pandas` - Enables us to manipulate imported datasets saved as dataframes.
- `Numpy`  - Enables us to compute mathematical functions as well as perform array operations.
- `Seaborn` - Enables us to visualise the data.
- `Matplotlib` - Additional library to assist in visualising the data.
- `Scipy.stats` - Assists in performing statistical calculations.
- `Sqlite3` - Allows us to perform SQL operations and access database data in this Jupyter notebook.

In [1]:
# importing relevant libraries
import pandas as pd 
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from scipy import stats
import sqlite3


%matplotlib inline

### Loading  and Review of Datasets  Individually

Once we have imported the libraries, next step is to load and review the datasets individually. This will help determine the datasets relevant to our analysis. 


### Rotten Tomatoes Movie Review Dataset


In [2]:
# Loading the Review tsv dataset display first and last 5 rows
df_review= pd.read_csv("zippedData/rt.reviews.tsv.gz", delimiter='\t', encoding='ISO-8859-1')
df_review.head()

id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   

           critic  top_critic         publisher               date  
0      PJ Nabarro           0   Patrick Nabarro  November 10, 2018  
1  Annalee Newitz           0           io9.com       May 23, 2018  
2    Sean Axmaker           0  Stream on Demand    January 4, 2018  
3   Daniel Kasman           0              MUBI  November 16, 2017  
4             NaN           0      Cinema Scope   October 12, 2017

From the dataframe above `df_review`, we seem to be missing crucial data relating to movies such as movie title, genre etc. It's quite difficult to make out the movie being critiqued based on the `review` and `id` column. Inaddition, for the columns that do have data, there seems to be incomplete data such as name of the critics and movie rating. Due to the inability to deduce the movie being critiqued, we will not be proceeding forward with this **Rotten Tomatoes Movie Review** dataset.

### Movie DB Dataset

In [4]:
# Load the Movie Db Dataset and Display the First and Last Five Rows
moviesdb_df= pd.read_csv("zippedData/tmdb.movies.csv.gz", compression='gzip')
moviesdb_df.head()

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

The `moviesdb` dataframe does provides us with a lot more information than the `df_review` dataframe. We are able to get the movie titles, popularity,vote count and cote average which may be quite beneficial in our analysis. However, the `genre` column is a list of numbers which requre additional mapping that we do not have access to. Due to this, we will not be utilizing this **Movie DB** dataset for our analysis.

### Movie_Info Dataset

In [5]:
# Loading movie_info tsv and Display the First and Last Five Rows
movie_info_df= pd.read_csv("zippedData/rt.movie_info.tsv.gz", sep='\t', compression='gzip')
movie_info_df.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

The `movie_info_df` dataframe possesses similar characteristics to the `df_review` dataframe. It does not give us information relating to the movie title and we are not able to link the id provided here to any other dataset. Also, the data has a lot of null values and as such we will not be utilizing the **Movie Info** dataset going forward. 

### Movie Budgets Dataset

In [29]:
# Loading movie budget tsv and Display the First and Last Five Rows
df_budget= pd.read_csv("zippedData/tn.movie_budgets.csv.gz")
df_budget.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

This `df_budget` dataframe provides a summary overview of the movie titles, release date as well as their corresponding budget and gross sales amounts. Thus, the **Movie Budgets** dataset will be quite beneficial to our analysis in the calculation of the return on investment and as such, we will proceed forward with it.

### Bom Movie Gross Dataset From Box office

In [6]:
bom_gross_df =pd.read_csv("zippedData/bom.movie_gross.csv.gz")
bom_gross_df.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

From the `bom_gross_df`, we get to have a view of the list of the box office movies along with the year of releases together with their corresponding gross sales amounts. As this analysis is focusing on performance of movies at the box office, this **BOM MOVIE** dataset will be our base dataset on which we will we center our analysis on.

### Loading SQL Database and creating a connection

Having loaded and reviewed the datasets, we now need to check if we can get additional data from the im.db database that we can build our anlaysis on. As such we will follow the following steps:

- Access the database through creating a connection using `sqlite3`
- Obtain the table names and review each of the tables in the database to determine which tables to retain for our analysis

In [8]:
#Access the Database

conn = sqlite3.connect("zippedData/im.db")

In [9]:
#Access the table names

table_name_query = """SELECT name 
                      AS 'Table Names' 
                      FROM sqlite_master 
                      WHERE type='table';"""

pd.read_sql(table_name_query, conn)

Table Names
0   movie_basics
1      directors
2      known_for
3     movie_akas
4  movie_ratings
5        persons
6     principals
7        writers

On review of tables in the database, we were able to drill down to the following four tables in the database:

- `directors` - Contains data on the director id associated with a particular movie ID 
- `persons` - Contains additional information relating to the director such as director name, occupation etc. 
- `movie_ratings` - Contains data relating to moviees such as runtime, genre, movie titles etc.
- `movie_basics` - Contains data relating to ratings, votecounts etc 

Therefore, we will now run a query that joins all these tables and extracts the following fields required for our analysis:

- `original_title`
- `start_year`
- `runtime_minutes`
- `genres`
- `averagerating`
- `numvotes`
- `primary_name`
-`primary_profession`



In [10]:
# Selecting key information for analysis from multiple tables in the database, including 
# movie titles, release years, runtime, genres, ratings, vote counts, director names, and
# professions. 
# This query integrates data from `movie_basics`, `movie_ratings`, `directors`, and `persons
# tables to align with project objectives.
df_imdb =pd.read_sql("""

SELECT DISTINCT original_title, 
                start_year,
                runtime_minutes,
                genres,
                averagerating,
                numvotes,
                primary_name,
                primary_profession
  FROM movie_basics
       JOIN movie_ratings
       ON movie_basics.movie_id = movie_ratings.movie_id
              JOIN directors
               ON movie_ratings.movie_id = directors.movie_id
                  JOIN persons
                    ON directors.person_id = persons.person_id;
 

""",conn)

df_imdb

original_title  start_year  runtime_minutes  \
0                       Sunghursh        2013            175.0   
1                 Ashad Ka Ek Din        2019            114.0   
2      The Other Side of the Wind        2018            122.0   
3                 Sabse Bada Sukh        2018              NaN   
4           La Telenovela Errante        2017             80.0   
...                           ...         ...              ...   
86019            Diabolik sono io        2019             75.0   
86020           Sokagin Çocuklari        2019             98.0   
86021                   Albatross        2017              NaN   
86022  La vida sense la Sara Amat        2019              NaN   
86023                  Drømmeland        2019             72.0   

                     genres  averagerating  numvotes          primary_name  \
0        Action,Crime,Drama            7.0        77   Harnam Singh Rawail   
1           Biography,Drama            7.2        43             Mani Kaul   
2                     Drama            6.9      4517          Orson Welles   
3              Comedy,Drama            6.1        13  Hrishikesh Mukherjee   
4      Comedy,Drama,Fantasy            6.5       119            Raoul Ruiz   
...                     ...            ...       ...                   ...   
86019           Documentary            6.2         6       Giancarlo Soldi   
86020          Drama,Family            8.7       136     Ahmet Faik Akinci   
86021           Documentary            8.5         8          Chris Jordan   
86022                  None            6.6         5             Laura Jou   
86023           Documentary            6.5        11    Joost van der Wiel   

                    primary_profession  
0             director,writer,producer  
1                director,writer,actor  
2                actor,director,writer  
3               director,editor,writer  
4             director,writer,producer  
...                                ...  
86019         director,writer,producer  
86020                  director,writer  
86021           director,writer,editor  
86022   miscellaneous,actress,director  
86023  director,cinematographer,writer  

[86024 rows x 8 columns]

### Merging the datasets

Now that we have all the datasets, next step is to merge them into one dataset and perform data cleaning on the merged datasets. The datasets we will be merging are:

- `bom_gross_df`
- `df_budget`
- `df_imdb`

We'll start off by merging `bom_gross_df` and `df_imdb` on the movie title column as it is common between the two dataframes. However, we will have to rename these columns in the their respective dataframes.

In [11]:
# We rename the common column in both dataFrames
bom_gross_df.rename(columns ={"title":"Movie_title"}, inplace = True)
df_imdb.rename(columns ={"original_title":"Movie_title"}, inplace = True)

In [12]:
# Confirming that the renaming has been updated in  bom_gross_df
bom_gross_df.head()

Movie_title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [13]:
# Confirming that the renaming has been updated in the df_imdb
df_imdb.head()

Movie_title  start_year  runtime_minutes  \
0                   Sunghursh        2013            175.0   
1             Ashad Ka Ek Din        2019            114.0   
2  The Other Side of the Wind        2018            122.0   
3             Sabse Bada Sukh        2018              NaN   
4       La Telenovela Errante        2017             80.0   

                 genres  averagerating  numvotes          primary_name  \
0    Action,Crime,Drama            7.0        77   Harnam Singh Rawail   
1       Biography,Drama            7.2        43             Mani Kaul   
2                 Drama            6.9      4517          Orson Welles   
3          Comedy,Drama            6.1        13  Hrishikesh Mukherjee   
4  Comedy,Drama,Fantasy            6.5       119            Raoul Ruiz   

         primary_profession  
0  director,writer,producer  
1     director,writer,actor  
2     actor,director,writer  
3    director,editor,writer  
4  director,writer,producer

As part of this step we observe the dimensions as well as presence of duplicates in both these dataframes before merging them.

In [16]:
#Checking the dimensions of the bom_gross_df

bom_gross_df.shape

(3387, 5)

In [17]:
#Checking the dimensions of the bom_gross_df

df_imdb.shape

(86024, 8)

In [22]:
# Checking for duplicates in bom_gross_df and df_imdb based on movie title
duplicates_bom_gross_df = bom_gross_df[bom_gross_df.duplicated(subset="Movie_title")]
duplicates_df_imdb = df_imdb[df_imdb.duplicated(subset="Movie_title")]
print("Duplicates in bom_gross_df =", len(duplicates_bom_gross_df))
print("Duplicates in df_imdb =" ,len(duplicates_df_imdb))

Duplicates in bom_gross_df = 1
Duplicates in df_imdb = 15637


In [24]:
# Observing the duplicate rows in the bom_gross_df 
duplicates_bom_gross_df.head()

Movie_title studio  domestic_gross foreign_gross  year
3045   Bluebeard  WGUSA         43100.0           NaN  2017

In [25]:
# Observing the duplicate rows in the bom_gross_df 
duplicates_df_imdb.head()

Movie_title  start_year  runtime_minutes  \
5                    La Telenovela Errante        2017             80.0   
20                            Gangavataran        2018            134.0   
27                          Kurzzeithelden        2015             45.0   
33  Quantum Quest: A Cassini Space Odyssey        2010             45.0   
62                       The Three Stooges        2012             92.0   

                        genres  averagerating  numvotes  \
5         Comedy,Drama,Fantasy            6.5       119   
20                        None            6.6         8   
27                      Sci-Fi            6.6        16   
33  Adventure,Animation,Sci-Fi            5.1       287   
62               Comedy,Family            5.1     28570   

               primary_name               primary_profession  
5         Valeria Sarmiento           editor,director,writer  
20  Dhundiraj Govind Phalke  director,writer,cinematographer  
27            Roman Gonther   visual_effects,director,writer  
33        Harry 'Doc' Kloor    producer,writer,miscellaneous  
62           Peter Farrelly         producer,writer,director

In [14]:
#Merge the two dataframes on a common feature Movie_title and create a new dataframe

merged_movies_df = bom_gross_df.merge(df_imdb, on = 'Movie_title', how = 'left')

We now proceed with a left merge and create a new dataframe `merged_movies_df`


In [15]:
# checking the shape number of rows and columns our new dataframe
merged_movies_df.shape

(4023, 12)

From the merge above, it does seem that there has been additional records added to our original `bom_gross_df` dataframe. This is expected if there is more than one record in the `df_imdb` that relate to a record in the `bom_gross_df` dataframe. 

In [18]:
# check the first few rows of the merged dataframe
merged_movies_df.head()

Movie_title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  start_year  runtime_minutes  \
0     652000000  2010      2010.0            103.0   
1     691300000  2010         NaN              NaN   
2     664300000  2010         NaN              NaN   
3     535700000  2010      2010.0            148.0   
4     513900000  2010      2010.0             93.0   

                       genres  averagerating   numvotes       primary_name  \
0  Adventure,Animation,Comedy            8.3   682218.0        Lee Unkrich   
1                         NaN            NaN        NaN                NaN   
2                         NaN            NaN        NaN                NaN   
3     Action,Adventure,Sci-Fi            8.8  1841066.0  Christopher Nolan   
4  Adventure,Animation,Comedy            6.3   167532.0      Mike Mitchell   

                          primary_profession  
0  editorial_department,editor,miscellaneous  
1                                        NaN  
2                                        NaN  
3                   writer,producer,director  
4                      actor,director,writer

Now that we have created a mereged dataframe `merged_data_frame`, we will proceed with merging this dataset with the `df_budget`dataframe as shown below. Similar steps will be applied here.

In [30]:
# Renaming movie column in df4 to have a commom feature we can merge on
df_budget.rename(columns ={"movie":"Movie_title"}, inplace = True)
df_budget.head()

id  release_date                                  Movie_title  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [31]:
# Merging df_merged with df_budget to create a comprehensive master dataset for analysis

final_merged_movies_df = merged_movies_df.merge(df_budget, on = 'Movie_title', how = 'left')
final_merged_movies_df.head()

Movie_title studio  domestic_gross_x  \
0                                  Toy Story 3     BV       415000000.0   
1                   Alice in Wonderland (2010)     BV       334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB       296000000.0   
3                                    Inception     WB       292600000.0   
4                          Shrek Forever After   P/DW       238700000.0   

  foreign_gross  year  start_year  runtime_minutes  \
0     652000000  2010      2010.0            103.0   
1     691300000  2010         NaN              NaN   
2     664300000  2010         NaN              NaN   
3     535700000  2010      2010.0            148.0   
4     513900000  2010      2010.0             93.0   

                       genres  averagerating   numvotes       primary_name  \
0  Adventure,Animation,Comedy            8.3   682218.0        Lee Unkrich   
1                         NaN            NaN        NaN                NaN   
2                         NaN            NaN        NaN                NaN   
3     Action,Adventure,Sci-Fi            8.8  1841066.0  Christopher Nolan   
4  Adventure,Animation,Comedy            6.3   167532.0      Mike Mitchell   

                          primary_profession    id  release_date  \
0  editorial_department,editor,miscellaneous  47.0  Jun 18, 2010   
1                                        NaN   NaN           NaN   
2                                        NaN   NaN           NaN   
3                   writer,producer,director  38.0  Jul 16, 2010   
4                      actor,director,writer  27.0  May 21, 2010   

  production_budget domestic_gross_y worldwide_gross  
0      $200,000,000     $415,004,880  $1,068,879,522  
1               NaN              NaN             NaN  
2               NaN              NaN             NaN  
3      $160,000,000     $292,576,195    $835,524,642  
4      $165,000,000     $238,736,787    $756,244,673

In [34]:
# Checking the shape of our master data
print(f"We have {final_merged_movies_df.shape[0]} rows and {final_merged_movies_df.shape[1]} columns")

We have 4043 rows and 17 columns


In [35]:
#Checking for dulicates based on movie entries

duplicates_final_merged_movies_df = final_merged_movies_df[final_merged_movies_df.duplicated(subset="Movie_title")]
print(len(duplicates_final_merged_movies_df ))

657


In [36]:
# Getting to understand the structure of our master data through a concise summary
final_merged_movies_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4043 entries, 0 to 4042
Data columns (total 17 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Movie_title         4043 non-null   object 
 1   studio              4037 non-null   object 
 2   domestic_gross_x    4008 non-null   float64
 3   foreign_gross       2466 non-null   object 
 4   year                4043 non-null   int64  
 5   start_year          2792 non-null   float64
 6   runtime_minutes     2727 non-null   float64
 7   genres              2787 non-null   object 
 8   averagerating       2792 non-null   float64
 9   numvotes            2792 non-null   float64
 10  primary_name        2792 non-null   object 
 11  primary_profession  2792 non-null   object 
 12  id                  1589 non-null   float64
 13  release_date        1589 non-null   object 
 14  production_budget   1589 non-null   object 
 15  domestic_gross_y    1589 non-null   object 
 16  worldw

We now have a master dataset that has merged all the datasets. Based on the `movie title` we can observe that we have 657 duplicates in this dataset already. Also, from the summary above, there seems to be a lot of null values in the columns following the merge. Therefore, the dataset needs to be cleaned before any meaningful, analysis can be done.

### Data Cleaning and Checking for Missing Values

To begin, we will make a copy of the data and make changes to this. This will ensure we have the original masterdata in the event we need to go back to it for additional reference



In [37]:
# Make a copy before modifying the masterdata
merged_movies_analysis_df= final_merged_movies_df.copy()

#### Check For Missing Values

As has been observed earlier, there are a lot of missing values. Let's take a deep look into this.

In [38]:
merged_movies_analysis_df.isna().sum()

Movie_title              0
studio                   6
domestic_gross_x        35
foreign_gross         1577
year                     0
start_year            1251
runtime_minutes       1316
genres                1256
averagerating         1251
numvotes              1251
primary_name          1251
primary_profession    1251
id                    2454
release_date          2454
production_budget     2454
domestic_gross_y      2454
worldwide_gross       2454
dtype: int64

 ##### Percentage of Null values
 

In [40]:
# Calculate null percentage for each column
null_percentage = [(col, (merged_movies_analysis_df[col].isna().sum() / len(merged_movies_analysis_df)) * 100) for col in merged_movies_analysis_df.columns]

# Create a DataFrame from the null percentages
null_percentage_df = pd.DataFrame(null_percentage, columns=["Columns", "Null Percentages"])

# Sort in descending order
null_percentage_df = null_percentage_df.sort_values(by="Null Percentages", ascending=False)
null_percentage_df

Columns  Null Percentages
16     worldwide_gross         60.697502
15    domestic_gross_y         60.697502
14   production_budget         60.697502
13        release_date         60.697502
12                  id         60.697502
3        foreign_gross         39.005689
6      runtime_minutes         32.550087
7               genres         31.066040
10        primary_name         30.942370
11  primary_profession         30.942370
8        averagerating         30.942370
9             numvotes         30.942370
5           start_year         30.942370
2     domestic_gross_x          0.865694
1               studio          0.148405
4                 year          0.000000
0          Movie_title          0.000000

##### Working with a dataset containing columns with 60% nulls can be challenging, but retaining these columns can be advantageous if they hold significant relevance to our business problem. Here’s why keeping these columns but working with fewer rows can be beneficial:

* Preserving Feature Diversity: Retaining column structure while using fewer rows maintains a broader set of features, potentially leading to a more comprehensive and balanced analysis.

* Enhancing Model Relevance: Retaining high-relevance columns (even with some missing data) can improve model accuracy, especially if these columns strongly influence the target variable. This approach may yield more relevant predictions compared to models with higher row counts but less meaningful features.

* Maintaining Critical Information: If certain columns (e.g., production budget, domestic/foreign revenue, runtime, or genre) are vital to the business question, dropping them would mean losing key metrics or context essential to achieving valuable insights.

* Avoiding Bias from Selective Imputation: Imputing values in high-null columns can introduce biases, especially if the missing data isn't randomly distributed. By carefully retaining rows rather than using selective imputation, the analysis remains more reliable and less prone to misleading conclusions.

By prioritizing relevant columns over row count, the dataset remains aligned with the primary analytical objectives, ultimately supporting more accurate and insightful outcomes while minimizing the risk of bias.



In [41]:
# Drop rows with null values and reassign to df5
merged_movies_analysis_df= merged_movies_analysis_df.dropna()

# Check if there are any remaining null values
merged_movies_analysis_df.isna().sum()

Movie_title           0
studio                0
domestic_gross_x      0
foreign_gross         0
year                  0
start_year            0
runtime_minutes       0
genres                0
averagerating         0
numvotes              0
primary_name          0
primary_profession    0
id                    0
release_date          0
production_budget     0
domestic_gross_y      0
worldwide_gross       0
dtype: int64

##### Check for Duplicates

We will now check to see if there are any duplicates present in this data. As a result of removing the null values, we can observe that the data is now free of any duplicates. We are now ready to proceed to perform analysis using this data.

In [42]:
# Checking for duplicate
print(f"We have {merged_movies_analysis_df.duplicated().sum()} duplicates in this dataset")

We have 0 duplicates in this dataset


In [43]:
# checking how many rows and columns in filtered df
print(f"We have {merged_movies_analysis_df.shape[0]} rows and {merged_movies_analysis_df.shape[1]} in our cleaned dataframe")

We have 1264 rows and 17 in our cleaned dataframe
